pypsrp/client.py

```diff
@@ -51,7 +51,7 @@ def __init__(self, server, **kwargs):
         """
         self.wsman = WSMan(server, **kwargs)
 
-    def copy(self, src, dest, configuration_name=DEFAULT_CONFIGURATION_NAME):
+    def copy(self, src, dest, expand_variables="", configuration_name=DEFAULT_CONFIGURATION_NAME):
```

Comment:
I feel like we should just simplify things and accept 1 style of env var substitution and have a kwarg that is just a bool (default to `False`). If we went down this route, the next question would be whether we accept the cmd style `%VAR%` or ps style `$env:VAR`. The former is quite easy to expand with `[System.Environment]::ExpandEnvironmentVariables($path)` but it is not really PowerShell like. Using the PowerShell style makes things conform to a similar language but we would need to find a way to inject code into the copy operation like `copy("", "$(Remove-Item C:\Windows -Force -Recurse)"), expand_variables="ps")`. We definitely want to limit this to just evaluating variables but not executing code. I'm not 100% sure whether this is easily done which is why I think the cmd style is easier to implement, just not really PowerShell like.

What are you're thoughts on this?

Reply:
I think that keeping the cmd-like format is enough. I guess that it's more common anyway and it will indeed simplify things.
What about the `expanduser` functionality for src though, are we okay with "not replicating" this functionality to the destination string? I will update the pr to change only the cmdlike format

Reply:
> What about the expanduser functionality for src though

That's a tricky question which I don't have an easy answer to. It seems like we can either

1. Keep things as is and always expand the client side paths using the Python code
2. Tie it to `expand_variables` so both the src/dest are expanded based on that var
3. Introduce another kwarg to control each expansion behaviour

I'm not a fan of 3, 2 would be a breaking change but I can see the merits of it and 1 is really just business as usual. Do you have any thoughts as to what way we should go here?

Reply:
To be honest I did not really realize that this will be a breaking change. The number 1 will keep it backward compatible but seems illogical to me.

When I do number 2 and expand is `False` by default, then it will break currently working scripts for some people. I did not realize that before and I don't think it's the right way anymore.

On the other hand, when expand is `True` by default, I think that it's less likely to break scripts for someone. Or do you think that there can be a case when people have `%VAR%` in a path and they don't want to have it expanded? I just don't see the use case. Also if it's some "random" text and not a real variable then it won't be expanded but the original string will be used.
Btw maybe I could add some check to only try the expansion if the path contains at least 2 `%`, to reduce the number of called remote commands.

Number 3 feels "overcomplicated" and I doubt that there would be someone to actually want to have one path expanded and not the other one :)
On the other hand I would go for this option if you think that expanding both src and dst by default is breaking too much. It's simple and straightforward